### PR TITLE
Fix #2090: avoid bang bang operator in networkinterceptor

### LINKED ISSUE #2090: use higher order functions for null check in NetworkInterceptor
current code uses bang bang operator though with null check 

Screenshot 2020-11-06 at 8 50 39 AM

we can use let higher order function to avoid this.

### PR DESCRIPTION
**Pull request by prudhvir3ddy**
## Explanation
removed bang bang operator which may null pointer exception though current code got null check but this is just a better way of coding and making use of kotlin higher order function . fixes #2090

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by anandwana001** [comment]
@prudhvir3ddy 
We recommend to follow few points:
1. PR title `Fix #bugnum: short description`
2. Branch naming, please check our wiki here - https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change
3. As per the last point in the above checklist, you can also mention me or @rt4914 for review in the PR description, until you get the Reviewer/Assignee section access.

**Comment by prudhvir3ddy** [comment]
@anandwana001 anyhow my branch will be deleted after this merge right ?

**Comment by anandwana001** [comment]
@prudhvir3ddy The branch is from your fork, so it's up to you to delete or not. As per the naming, you can choose a descriptive naming with all small caps with `-` to separate, from your next PR's.

File: data/src/main/java/org/oppia/android/data/backends/gae/NetworkInterceptor.kt
Comment (aggarwalpulkit596) [inline data/src/main/java/org/oppia/android/data/backends/gae/NetworkInterceptor.kt]: Instead of using `it` could you something more descriptive maybe
`resBody` or simply  `body`

**Review by aggarwalpulkit596 — CHANGES_REQUESTED** [review]

**Review by aggarwalpulkit596 — APPROVED** [review]

**Comment by aggarwalpulkit596** [comment]
LGTM @anandwana001 do you have anything else or should i go ahead and merge this one.

**Comment by anandwana001** [comment]
@rt4914 I will defer to Rajat. Please also fill the contributor survey to take contributions further. 
Also, checks are failing.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks. @anandwana001 You can make a note of this as you are the onboarding mentor.

**Comment by anandwana001** [comment]
@rt4914  Thanks

**Pull request merged**
